### ISSUE TITLE
codegen: warn on pg default which isn't added to joist

### ISSUE THREAD

**Issue by brudil** [joist-orm/joist-orm#1258]
Not sure if there would be false positives with this - can't think of any right now.

codegen warn on any fields which have a pg default define which joist ignores (such as jsonb fields)

**Comment by stephenh** [comment]
Ah nice!

Are you thinking that if the user adds an explicit `config.setDefault("fieldWithThePgOnlyDefault", ...)`, that would silence the notification?

**Comment by brudil** [comment]
Correct. atm I'm seeing it as a dx gap where joist uses attempts to use defaults, but unless the user is aware of what joist can and can't pick up there, there's no feedback to say "this might be a default, but it's not a default in Joist yet"

**Comment by stephenh** [comment]
Yep, that makes sense.

Only question would be if there is a PG default that, besides the joist codegen not being able to translate, is just hard to express in code as well, what would the user do, to at least silence the warning?

Not sure I have any good examples of "pg defaults that would be hard to express in typescript"...maybe something that uses db functions, etc...
